src/kernels/kernelProvider.node.ts

```diff
@@ -60,16 +58,6 @@ export class KernelProvider extends BaseCoreKernelProvider {
 
         const resourceUri = notebook.notebookType === InteractiveWindowView ? options.resourceUri : notebook.uri;
         const settings = createKernelSettings(this.configService, resourceUri);
-        const kernelExecution = new KernelExecution(
```

Comment:
Kernels don't need to know anything about Execution
This PR now removes the cyclic dependency